packages/token-dictionary/src/token.ts

```diff
@@ -121,8 +152,11 @@ export class Token {
       if (referenceToken.isConditional) {
         return valueStr
       }
+
       const value = referenceToken.expandReferences()
-      return valueStr.replace(`{${key}}`, value)
+      this.setRawValue()
+
+      return valueStr.replaceAll(`{${key}}`, value)
```

Comment:
@segunadebayo @astahmer 
I've changed `replace` to `replaceAll` in cases you need to repeat the same token twice, for example:
```
const tokens = {
  colors: {
    gradient: {
       value: 'linear-gradient(90deg, {colors.blue.400} 0%, {colors.blue.100} 35%, {colors.blue.400} 100%)',
    },
 },
};
```
This was not a part of the original issue.